src/main/java/com/aws/greengrass/deployment/exceptions/RetryableGGDeploymentException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.deployment.exceptions;
+
+import com.aws.greengrass.deployment.errorcode.DeploymentErrorCode;
+
+/**
+ * Exception for handling 5xx deployment failures.
+ */
+public class RetryableGGDeploymentException extends DeploymentException {
+    static final long serialVersionUID = -3387516993124229948L;
+
+    public RetryableGGDeploymentException(String message) {
+        super(message);
+    }
+
+    public RetryableGGDeploymentException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public RetryableGGDeploymentException(Throwable throwable) {
+        super(throwable);
+    }
+
+    public RetryableGGDeploymentException(String message, DeploymentErrorCode errorCode) {
+        super(message);
+        super.addErrorCode(errorCode);
+    }
+
+    @Override
+    public RetryableGGDeploymentException withErrorContext(Throwable t, DeploymentErrorCode errorCode) {
+        super.withErrorContext(t, errorCode);
+        return this;
+    }
```

Comment:
Need to also verify that the correct error code is reported when a deployment actually fails with a RetryableGGDeploymentException (after infinite retry)

Reply:
How can a deployment fail if it is being retried indefinitely? It can be canceled, but that's not a failure

Reply:
that's true for most cases.

looks like the only place we retry finitely is local deployment listThingGroupsForCoreDevice - https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java#L192-L216. 

I think in that case, we should add RetryableGGDeploymentException in the list `DEVICE_OFFLINE_INDICATIVE_EXCEPTIONS` and proceed the local deployment

yea error code is probably not needed